src/main/java/org/jenkinsci/plugins/workflow/steps/CoreWrapperStep.java

```diff
@@ -92,20 +94,82 @@ public SimpleBuildWrapper getDelegate() {
         }
 
         private void doStart() throws Exception {
-            SimpleBuildWrapper.Context c = new SimpleBuildWrapper.Context();
-            Run<?, ?> run = getContext().get(Run.class);
-            delegate.setUp(c, run, getContext().get(FilePath.class), getContext().get(Launcher.class), getContext().get(TaskListener.class), getContext().get(EnvVars.class));
-            BodyInvoker bodyInvoker = getContext().newBodyInvoker();
+            SimpleBuildWrapper.Context c = null;
+            // In Jenkins 2.25x, a createContext() method on SimpleBuildWrapper is required to ensure that a Disposer
+            // registered on that context inherits the wrapper's workspace requirement. Use it when available.
+            // TODO: Use 'c = this.delegate.createContext()' once this plugin depends on Jenkins 2.25x or later.
+            try {
+                final Method createContext = SimpleBuildWrapper.class.getMethod("createContext");
+                c = (SimpleBuildWrapper.Context) createContext.invoke(this.delegate);
+            }
+            catch (NoSuchMethodException e) {
+                c = new SimpleBuildWrapper.Context();
+            } catch (InvocationTargetException ite) {
+                final Throwable realException = ite.getCause();
+                throw realException instanceof Exception ? (Exception) realException : ite;
+            }
+            final StepContext context = getContext();
+            final Run<?, ?> run = context.get(Run.class);
+            assert run != null;
+            {
+                final TaskListener listener = context.get(TaskListener.class);
+                assert listener != null;
+                final EnvVars env = context.get(EnvVars.class);
+                assert env != null;
+                final FilePath workspace = context.get(FilePath.class);
+                final Launcher launcher = context.get(Launcher.class);
+                boolean workspaceRequired = true;
+                // In Jenkins 2.25x, a SimpleBuildWrapper can indicate that it does not require a workspace context by
+                // overriding a requiresWorkspace() method to return false. So use that if it's available.
+                // Note: this uses getMethod() on the delegate's type and not SimpleBuildWrapper so that an
+                // implementation can get this behaviour without switching to Jenkins 2.25x itself.
+                // TODO: Use 'workspaceRequired = this.delegate.requiresWorkspace()' once this plugin depends on Jenkins 2.25x or later.
```

Comment:
```suggestion
                // In Jenkins 2.258, a SimpleBuildWrapper can indicate that it does not require a workspace context by
                // overriding a requiresWorkspace() method to return false. So use that if it's available.
                // Note: this uses getMethod() on the delegate's type and not SimpleBuildWrapper so that an
                // implementation can get this behaviour without switching to Jenkins 2.258 itself.
                // TODO: Use 'workspaceRequired = this.delegate.requiresWorkspace()' once this plugin depends on Jenkins 2.258 or later.
```